addons/account/views/account_move_views.xml

```diff
@@ -163,15 +163,15 @@
             <field name="model">account.move.line</field>
             <field eval="100" name="priority"/>
             <field name="arch" type="xml">
-                <tree string="Journal Items" create="false" edit="false" expand="context.get('expand', False)" multi_edit="1" sample="1">
+                <tree string="Journal Items" create="false" expand="context.get('expand', False)" multi_edit="1" sample="1">
                     <field name="date" readonly="1"/>
                     <field name="company_id" invisible="1"/>
                     <field name="company_id" groups="base.group_multi_company" optional="hide"/>
                     <field name="journal_id" options='{"no_open":True}' optional="hide"/>
                     <field name="move_name" string="Journal Entry" widget="open_move_widget"/>
                     <field name="account_id" options="{'no_open': True, 'no_create': True}" domain="[('company_id', '=', company_id)]" groups="account.group_account_readonly"/>
                     <field name="partner_id" optional="show" attrs="{'readonly':[('account_type','in', ('liability_payable','asset_receivable'))]}"/>
```

Comment:
on sales & purchase journals only

Reply:
> on sales & purchase journals only

How do you do that, there is only one view now...

Reply:
Is something like this `{'readonly':['|', '&amp;', ('account_type', 'in', ('liability_payable','asset_receivable')), ('matching_number', '!=', False),  ('move_type', '!=', 'entry')]}` what you thought @william-andre ?

Reply:
No, I would make the domain here only based on the move type, and check in `write` if the line has a partial before trying to change the partner. (allow it in the UI but block if we can't)

Reply:
Is it better like it is now ?

Reply:
Yup!